File: Section Reports/Preview/CustomPreview/C#/CustomPreview/PreviewForm.cs
using System;
using System.Drawing;
using System.ComponentModel;
using System.Windows.Forms;
using System.IO;
namespace GrapeCity.ActiveReports.Samples.CustomPreview
{
	/// <summary>
	/// PreviewForm - child MDI form that loads the ActiveReports Viewer to view a report
	/// and provides options to export, save and print the generated report.
	/// </summary>
	public partial class PreviewForm : System.Windows.Forms.Form
	{
		private IContainer components;
		public PreviewForm(Document.SectionDocument doc, Form parentForm)
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();
			
			MdiParent = parentForm;
			arvMain.LoadDocument(doc);
			Text = doc.Name;
		}
		private void PreviewForm_Load(object sender, System.EventArgs e)
		{
			// Add custom buttons to the viewer's toolbar.
			Icon ico = new Icon(GetType().Module.Assembly.GetManifestResourceStream("GrapeCity.ActiveReports.Samples.CustomPreview.Resources.Save16x16.ico"));
			arvMain.Toolbar.ToolStrip.ImageList
				= new ImageList();
			arvMain.Toolbar.ToolStrip.ImageList.Images.Add("saveicon", ico);
			ico = new Icon(GetType().Module.Assembly.GetManifestResourceStream("GrapeCity.ActiveReports.Samples.CustomPreview.Resources.Export16x16.ico"));
			arvMain.Toolbar.ToolStrip.ImageList.Images.Add("exporticon", ico);
			ToolStrip ts = arvMain.Toolbar.ToolStrip;
			//Add Save toolstrip button to the toolbar.
			
			ToolStripButton tsbSaveRdf = new ToolStripButton("Save");
			
			
			tsbSaveRdf.Image = arvMain.Toolbar.ToolStrip.ImageList.Images["saveicon"];
			tsbSaveRdf.Click += new EventHandler(tsbSaveRdf_Click);
			//Add Export toolstrip button to the toolbar
			
			ToolStripButton tsbExport = new ToolStripButton("Export");
			
			
			tsbExport.Image = arvMain.Toolbar.ToolStrip.ImageList.Images["exporticon"];
			tsbExport.Click += new EventHandler(tsbExport_Click);
			ts.Items.Add(new ToolStripSeparator());
			ts.Items.Add(tsbSaveRdf);
			ts.Items.Add(new ToolStripSeparator());
			ts.Items.Add(tsbExport);
		}
		void tsbExport_Click(object sender, EventArgs e)
		{
			ExportDocument();
		}
		void tsbSaveRdf_Click(object sender, EventArgs e)
		{
			SaveDocument();
		}
		
		/// <summary>
		/// Saves out an existing report document to the RDF format.
		/// </summary>
		private void SaveDocument()
		{
			SaveFileDialog dlgSave = new SaveFileDialog();
			
			dlgSave.Title = "Save Report Document";
			
			
			
			dlgSave.Filter = "Report Document Files (*.rdf)|*.rdf";
			
			
			dlgSave.DefaultExt = "rdf";
			dlgSave.AddExtension = true;
			if (dlgSave.ShowDialog(this) == DialogResult.OK)
			{
				if (File.Exists(dlgSave.FileName))
				{
					
					if (MessageBox.Show(this, "Overwrite Existing File?", "Overwrite", MessageBoxButtons.YesNo) != DialogResult.Yes)
					
					
					{
						return;
					}
				}
				arvMain.Document.Save(dlgSave.FileName);
			}
		}
		/// <summary>
		/// Opens the export form to export the current report document.
		/// </summary>
		private void ExportDocument()
		{
			ExportForm exportForm = new ExportForm(arvMain.Document);
			exportForm.ShowDialog(this);
		}
		/// <summary>
		/// Call ExportDocument() on mnuExport_Click to to export the
		/// current report document.
		/// </summary>
		private void mnuExport_Click(object sender, EventArgs e)
		{
			ExportDocument();
		}
		/// <summary>
		/// Call SaveDocument() on mnuSaveDocument_Click to save the 
		/// current report document to RDF format.
		/// </summary>
		private void mnuSaveDocument_Click(object sender, EventArgs e)
		{
			SaveDocument();
		}
	}
}

File: Section Reports/Preview/CustomPreview/VB.NET/CustomPreview/PreviewForm.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports System.IO
' PreviewForm - child MDI form that loads the ActiveReports Viewer to view a report
' and provides options to export, save and print the generated report.
Public Class PreviewForm
	Inherits Windows.Forms.Form
	
	Private WithEvents tsbSaveRdf As New ToolStripButton("Save")
	
	'
	
	Private WithEvents tsbExport As New ToolStripButton("Export")
	
	'
	' Required designer variable.
	Public Sub New(ByVal doc As Document.SectionDocument, ByVal parentForm As Form)
		MyBase.New()
		'Required for Windows Form Designer support
		InitializeComponent()
		MdiParent = parentForm
		arvMain.Document = doc
		Text = doc.Name
	End Sub
	'Clean up any resources being used.
	Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
		If disposing Then
			If Not (components Is Nothing) Then
				components.Dispose()
			End If
		End If
		MyBase.Dispose(disposing)
	End Sub
	Private Sub PreviewForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
		' Add custom buttons to the viewer's toolbar.
		Dim ico As New Icon([GetType]().[Module].Assembly.GetManifestResourceStream("GrapeCity.ActiveReports.Samples.CustomPreview.Save16x16.ico"))
		arvMain.Toolbar.ToolStrip.ImageList = New ImageList()
		arvMain.Toolbar.ToolStrip.ImageList.Images.Add("saveicon", ico)
		ico = New Icon([GetType]().[Module].Assembly.GetManifestResourceStream("GrapeCity.ActiveReports.Samples.CustomPreview.Export16x16.ico"))
		arvMain.Toolbar.ToolStrip.ImageList.Images.Add("exporticon", ico)
		Dim ts As ToolStrip = arvMain.Toolbar.ToolStrip
		'Add Save toolstrip button to the toolbar.
		tsbSaveRdf.Image = arvMain.Toolbar.ToolStrip.ImageList.Images("saveicon")
		'Add Export toolstrip button to the toolbar
		tsbExport.Image = arvMain.Toolbar.ToolStrip.ImageList.Images("exporticon")
		ts.Items.Add(tsbSaveRdf)
		ts.Items.Add(New ToolStripSeparator())
		ts.Items.Add(tsbExport)
	End Sub 'PreviewForm_Load
	Private Sub tsbExport_Click(ByVal sender As Object, ByVal e As EventArgs) Handles tsbExport.Click
		ExportDocument()
	End Sub
	Private Sub tsbSaveRdf_Click(ByVal sender As Object, ByVal e As EventArgs) Handles tsbSaveRdf.Click
		SaveDocument()
	End Sub
	'Saves out an existing report document to the RDF format.
	Private Sub SaveDocument()
		Dim dialogSave As New SaveFileDialog
		
		dialogSave.Title = "Save Report Document"
		
		'
		
		dialogSave.Filter = "Report Document Files (*.rdf)|*.rdf"
		
		'
		dialogSave.DefaultExt = "rdf"
		dialogSave.AddExtension = True
		If dialogSave.ShowDialog(Me) = DialogResult.OK Then
			If File.Exists(dialogSave.FileName) Then
				
				If MessageBox.Show(Me, "Overwrite Existing File?", "Overwrite", MessageBoxButtons.YesNo) <> DialogResult.Yes Then
					
					'
					Return
				End If
			End If
			arvMain.Document.Save(dialogSave.FileName)
		End If
	End Sub 'SaveDocument
	'Opens the export form to export the current report document.
	Private Sub ExportDocument()
		Dim exportForm As New ExportForm(arvMain.Document)
		exportForm.ShowDialog(Me)
	End Sub 'ExportDocument
	'Call ExportDocument() on mnuExport_Click to to export the
	' current report document.
	Private Sub mnuExport_Click(ByVal sender As Object, ByVal e As EventArgs) Handles mnuExport.Click
		ExportDocument()
	End Sub 'mnuExport_Click
	'Call SaveDocument() on mnuSaveDocument_Click to save the 
	' current report document to RDF format.
	Private Sub mnuSaveDocument_Click(ByVal sender As Object, ByVal e As EventArgs) Handles mnuSaveDocument.Click
		SaveDocument()
	End Sub 'mnuSaveDocument_Click
End Class 'PreviewForm
